pkg/config/config.go

```diff
@@ -586,3 +586,94 @@ func generateOrgConfigGroup(org *Organization, mspConfig *mb.MSPConfig) (*cb.Con
 
 	return orgGroup, nil
 }
+
+// CreateSignedConfigUpdateEnvelope creates a signed configuration update envelope.
+func CreateSignedConfigUpdateEnvelope(configUpdate *cb.ConfigUpdate, signingIdentity *SigningIdentity, signatures ...*cb.ConfigSignature) (*cb.Envelope, error) {
+	if configUpdate == nil {
+		return nil, errors.New("no config update specified")
+	}
+	if signingIdentity == nil {
+		return nil, errors.New("no signing identity specified")
+	}
+	if len(signatures) == 0 {
+		return nil, errors.New("no signatures specified")
+	}
+
+	update, err := proto.Marshal(configUpdate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal config update")
+	}
+
+	configUpdateEnvelope := &cb.ConfigUpdateEnvelope{
+		ConfigUpdate: update,
+		Signatures:   signatures,
+	}
+
+	signedEnvelope, err := createSignedEnvelopeWithTLSBinding(cb.HeaderType_CONFIG_UPDATE, configUpdate.ChannelId, signingIdentity, configUpdateEnvelope)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create signed config update envelope: %s", err)
+	}
+
+	return signedEnvelope, nil
+}
+
+// CreateSignedEnvelopeWithTLSBinding creates a signed envelope of the desired
+// type, with marshaled dataMsg and signs it. It also includes a TLS cert hash
+// into the channel header
+func createSignedEnvelopeWithTLSBinding(
+	txType cb.HeaderType,
+	channelID string,
+	signer *SigningIdentity,
+	envelope proto.Message,
+) (*cb.Envelope, error) {
+	channelHeader := &cb.ChannelHeader{
+		Type: int32(txType),
+		Timestamp: &timestamp.Timestamp{
+			Seconds: time.Now().Unix(),
+			Nanos:   0,
```

Comment:
nit: don't need to specify zero values...

there's also a helper in the ptypes package that does exactly this. https://github.com/golang/protobuf/blob/d23c5127dc24889085f8ccea5c9d560a57a879d8/ptypes/timestamp.go#L103